src/browser/BrowserService.cpp

```diff
@@ -686,14 +690,15 @@ BrowserService::searchEntries(const QSharedPointer<Database>& db, const QString&
 
             // Search for additional URL's starting with KP2A_URL
             for (const auto& key : entry->attributes()->keys()) {
-                if (key.startsWith(ADDITIONAL_URL) && handleURL(entry->attributes()->value(key), siteUrlStr, formUrlStr)
+                if (key.startsWith(ADDITIONAL_URL)
+                    && handleURL(entry->attributes()->value(key), siteUrlStr, formUrlStr, omitWwwSubdomain)
                     && !entries.contains(entry)) {
                     entries.append(entry);
                     continue;
                 }
             }
 
-            if (!handleEntry(entry, siteUrlStr, formUrlStr)) {
+            if (!handleEntry(entry, siteUrlStr, formUrlStr, omitWwwSubdomain)) {
                 continue;
```

Comment:
I feel like all of this for loop should actually be in the handleEntry function and use the `Entry::getAllUrls()` instead of rolling a new algorithm here. Additionally handleEntry should really be named `shouldHandleEntry` or `shouldIncludeEntry`.

Reply:
Done.